src/main/webapp/js/components/widgets/plot/Plot.js

```diff
@@ -646,6 +646,52 @@ define(function (require) {
 			return this;
 		},
 
+        genColorscale: function(min, max, n, f) {
+            var colorscale = [];
+            var step = (max-min)/n;
+            var x = min;
+            var rgb = [];
+            for (var i=0; i<n; ++i) {
+                colorscale.push([i/n, f(x)]);
+                colorscale.push([(i+1)/n, f(x)]);
+                x += step;
+            }
+
+            return colorscale;
+        },
+
+        setScale: function(min, max, scalefn, normalize){
```

Comment:
@mattearnshaw can you add some comments? What do these methods do? In what workflows are they used?

Reply:
these went here for persisting colorbar as data, they are really just naughty duplicates from colorbar.js in geppetto-osb. have removed.